pkg/ccl/changefeedccl/enriched_source_provider.go

```diff
@@ -87,76 +91,81 @@ func newEnrichedSourceData(
 
 func newEnrichedSourceProvider(
 	opts changefeedbase.EncodingOptions, sourceData enrichedSourceData,
-) *enrichedSourceProvider {
+) (*enrichedSourceProvider, error) {
+	jsonBase := map[string]json.JSON{
+		fieldNameJobID:              json.FromString(sourceData.jobID),
+		fieldNameChangefeedSink:     json.FromString(sourceData.sink),
+		fieldNameDBVersion:          json.FromString(sourceData.dbVersion),
+		fieldNameClusterName:        json.FromString(sourceData.clusterName),
+		fieldNameClusterID:          json.FromString(sourceData.clusterID),
+		fieldNameSourceNodeLocality: json.FromString(sourceData.sourceNodeLocality),
+		fieldNameNodeName:           json.FromString(sourceData.nodeName),
+		fieldNameNodeID:             json.FromString(sourceData.nodeID),
+	}
+
+	var nonFixedJSONFields []string
+	nonFixedDataIdx := map[string]int{}
+	if opts.MVCCTimestamps {
+		nonFixedJSONFields = append(nonFixedJSONFields, fieldNameMVCCTimestamp)
+		nonFixedDataIdx[fieldNameMVCCTimestamp] = len(nonFixedJSONFields) - 1
+	}
+	// TODO(#139661): Add other non fixed fields.
+
+	jpo, err := json.NewPartialObject(jsonBase, nonFixedJSONFields)
+	if err != nil {
+		return nil, err
+	}
+
 	return &enrichedSourceProvider{
 		sourceData: sourceData,
 		opts: enrichedSourceProviderOpts{
 			mvccTimestamp: opts.MVCCTimestamps,
 			updated:       opts.UpdatedTimestamps,
```

Comment:
`updated` appears to be unused. Could you add a TODO with an issue link to make sure we use it?

Reply:
this is covered by these TODOs: `// TODO(#139661): Add other non fixed fields, in the right order.`